File: app/src/main/java/com/example/mvvmexempel/MainActivity.java
package com.example.mvvmexempel;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import com.example.mvvmexempel.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    LoginViewModel loginViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this,R.layout.activity_main);
        loginViewModel = new ViewModelProvider(this).get(LoginViewModel.class);
        binding.setLoginViewModel(loginViewModel);
        binding.setLifecycleOwner(this);

        loginViewModel.getLoginResultLiveData().observe(this, new Observer<Boolean>() {
            @Override
            public void onChanged(Boolean isValid) {
                if(isValid){
//                    Toast.makeText(MainActivity.this,"inloggning lyckades", Toast.LENGTH_SHORT).show();
                    Intent intent = new Intent(MainActivity.this, MemberActivity.class);
                    startActivity(intent);
                }else{
                    Toast.makeText(MainActivity.this,"fel användarnamn / lösenord", Toast.LENGTH_SHORT).show();
                }
            }
        });

        loginViewModel.getregisteredResultLiveData().observe(this, new Observer<User>() {
            @Override
            public void onChanged(User user) {
                if (user != null){
                    String message = "User " + user.getUserName() + " registered.";
                    Toast.makeText(MainActivity.this,message, Toast.LENGTH_SHORT).show();
                }
            }
        });

    }
}

File: app/src/main/java/com/example/aniamtionexempel/MainActivity.kt
package com.example.aniamtionexempel

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.SurfaceView
import com.example.aniamtionexempel.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        //setContentView(GameView(this))

        setContentView(binding.root)
        val gameView = GameView(this)
        val container = binding.root
        container.addView(gameView)
//        binding.surfaceView.holder.addCallback(GameView(this))



    }
}